File: Backend/Models/GameStage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IdiomLearningAPI.Models
{
    /// <summary>
    /// 게임 스테이지 모델 (12지신)
    /// FR 4.1: 스테이지 맵
    /// </summary>
    [Table("GameStages")]
    public class GameStage
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int StageId { get; set; }

        [Required]
        [MaxLength(100)]
        public string BossName { get; set; } = string.Empty;

        public int BossHp { get; set; }

        public int BossAttackPower { get; set; }

        [MaxLength(500)]
        public string? BossImageUrl { get; set; }

        [Required]
        [MaxLength(100)]
        public string ZodiacAnimal { get; set; } = string.Empty;

        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        [MaxLength(10)]
        public string Emoji { get; set; } = string.Empty;

        [Required]
        public Difficulty RequiredDifficulty { get; set; }
    }
}

File: server/models/GameStage.js
const mongoose = require('mongoose');

const gameStageSchema = new mongoose.Schema({
  stage_id: {
    type: Number,
    required: true,
    unique: true,
    min: 1,
    max: 12,
  },
  boss_name: {
    type: String,
    required: true,
  },
  boss_hp: {
    type: Number,
    required: true,
  },
  boss_attack_power: {
    type: Number,
    required: true,
  },
  boss_image_url: {
    type: String,
    default: null,
  },
  zodiac_animal: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    default: '',
  },
});

module.exports = mongoose.model('GameStage', gameStageSchema);
